core/base/src/TFolder.cxx

```diff
@@ -311,41 +311,36 @@ TObject *TFolder::FindObject(const TObject *) const
 
 TObject *TFolder::FindObject(const char *name) const
 {
-   if (!fFolders) return 0;
-   if (name == 0) return 0;
+   if (!fFolders || !name) return nullptr;
    if (name[0] == '/') {
       if (name[1] == '/') {
-         if (!strstr(name,"//root/")) return 0;
+         if (!strstr(name,"//root/")) return nullptr;
          return gROOT->GetRootFolder()->FindObject(name+7);
       } else {
          return gROOT->GetRootFolder()->FindObject(name+1);
       }
    }
    Int_t nch = strlen(name);
-   char *cname;
    char csname[128];
-   if (nch < (int)sizeof(csname))
-      cname = csname;
-   else
-      cname = new char[nch+1];
-   strcpy(cname, name);
-   TObject *obj;
+   char *cname = csname;
+   Int_t len = sizeof(csname);
+   if (nch >= len) {
+      len = nch+1;
+      cname = new char[len];
+   }
+   strncpy(cname, name, len);
```

Comment:
Instead of `strncpy` I'd prefer the use of the superior [`strlcpy`](https://linux.die.net/man/3/strlcpy) which we make available in ROOT in a platform-independent way.